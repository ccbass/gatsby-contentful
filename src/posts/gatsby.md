---
title: "Basic Gatsby Tutorial"
date: "2021-08-28"
---

Here's the tutorial markdown post!

## Topics

1. Gatsby
2. GraphQL
3. React
