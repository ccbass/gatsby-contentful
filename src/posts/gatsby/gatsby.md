---
title: "Basic Gatsby Tutorial"
date: "2021-08-28"
---

Here's the tutorial markdown post!

![grass image](./grass.jpg)
## Topics

1. Gatsby
2. GraphQL
3. React
