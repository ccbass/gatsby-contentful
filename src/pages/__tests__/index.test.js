import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery } from "gatsby"
import IndexPage from "../index"
import Head from '../../components/head'


beforeEach(() => {
  jest.clearAllMocks();

  useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          author: `CCB`,
          title: 'Full-Stack Engineering Blog'
        },
      },
  }))
})

describe("Index Page Tests", () => {
  it("Renders as expected vs snapshot.", () => {
    const index = renderer.create(<IndexPage />).toJSON()
    expect(index).toMatchSnapshot()

  })

  it("Helmet has correct props.", () => {
    const index = renderer.create(<IndexPage />)
    const indexInstance = index.root

    expect(indexInstance.findByType(Head).props.title).toBe('Home');

  })

})
