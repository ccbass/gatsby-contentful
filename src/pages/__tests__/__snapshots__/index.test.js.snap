// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page Tests Renders as expected vs snapshot. 1`] = `
<div>
  <div>
    <header>
      <h1>
        <a
          href="/"
        >
          Full-Stack Engineering Blog
        </a>
      </h1>
      <nav>
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="/blog"
            >
              Blog
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <h1>
      Welcome.
    </h1>
    <h2>
      I'm Colin, an engineer based out of Oregon.
    </h2>
    <p>
      Need a dev? 
      <a
        href="/contact"
      >
        Contact Me
      </a>
    </p>
  </div>
  <footer>
    <p>
      Created by 
      CCB
      , Copyright 2021
    </p>
  </footer>
</div>
`;
